src/adapter/aws-lambda/handler.ts

```diff
@@ -22,6 +22,7 @@ export interface APIGatewayProxyEventV2 {
   version: string
   routeKey: string
   headers: Record<string, string | undefined>
+  multiValueHeaders?: undefined
```

Comment:
It allows 

```javascript
event.multiValueHeaders?.['host']?.[0]
```
instead of
```javascript
'multiValueHeaders' in event
      ? event.multiValueHeaders.['host']?.[0]
      : undefined
```

Reply:
Thank you. The implementation is clean, and I think this is good.

---

Note. 

Below is a note for when the need arises to completely remove 'multiValueHeaders'.

```ts
const createRequest = (event: LambdaEvent) => {
  const queryString = extractQueryString(event)
  const domainName =
    event.requestContext && 'domainName' in event.requestContext
      ? event.requestContext.domainName
      : getHostFromEvent(event)

  const path = isProxyEventV2(event) ? event.rawPath : event.path
  const urlPath = `https://${domainName}${path}`
  const url = queryString ? `${urlPath}?${queryString}` : urlPath

  const headers = new Headers()
  getCookies(event, headers)
  if (event.headers) {
    for (const [k, v] of Object.entries(event.headers)) {
      if (v) {
        headers.set(k, v)
      }
    }
  }
  if (isAPIGatewayProxyEvent(event) && event.multiValueHeaders) {
    for (const [k, values] of Object.entries(event.multiValueHeaders)) {
      if (values) {
        values.forEach((v) => headers.append(k, v))
      }
    }
  }

  const method = isProxyEventV2(event) ? event.requestContext.http.method : event.httpMethod
  const requestInit: RequestInit = {
    headers,
    method,
  }

  if (event.body) {
    requestInit.body = event.isBase64Encoded ? Buffer.from(event.body, 'base64') : event.body
  }

  return new Request(url, requestInit)
}

const getHostFromEvent = (event: LambdaEvent): string => {
  if (!event.headers) {
    return 'localhost'
  }

  if (isAPIGatewayProxyEvent(event)) {
    return event.headers['host'] ?? event.multiValueHeaders?.['host']?.[0] ?? 'localhost'
  } else {
    return event.headers['host'] ?? 'localhost'
  }
}
```

```ts
const isAPIGatewayProxyEvent = (event: LambdaEvent): event is APIGatewayProxyEvent => {
  return 'multiValueHeaders' in event
}
```